gson/src/main/java/com/google/gson/internal/reflect/ReflectionHelper.java

```diff
@@ -169,22 +177,34 @@ public <T> Constructor<T> getCanonicalRecordConstructor(Class<T> raw) {
       try {
         Object recordComponents = getRecordComponents.invoke(raw);
         int componentCount = Array.getLength(recordComponents);
-        Class[] recordFields = new Class[componentCount];
+        Class<?>[] recordComponentType = new Class<?>[componentCount];
```

Comment:
The name should probably be plural:
```suggestion
        Class<?>[] recordComponentTypes = new Class<?>[componentCount];
```